src/routes/reviewRoutes.js

```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+const router = express.Router();
+const { authenticateUser } = require('../middleware/authentication');
+
+const {
+  createReview,
+  updateReview,
+} = require('../controllers/reviewController');
+
+router.route('/album/:albumId').post(authenticateUser, createReview);
+
+//? for Akos- does it matter which type of request we add here- put or patch? because I used put but it didn't overwright the id of the review
+router.route('/:id').put(authenticateUser, updateReview);
```

Comment:
@akosasante so with put request we should pass in the req.body all the required fields and it will throw an error if we don't, right? and with patch we just pass the fields we need to update?

Reply:
put overwrights everything so if the fields are not required and we pass less properties than it had initially- it will fully replace the item in a db and we'll have only properties we passed in our put request.

Reply:
does 'upsert' means smth in between 'insert' and 'update'? @akosasante

Reply:
> @akosasante so with put request we should pass in the req.body all the required fields and it will throw an error if we don't, right? and with patch we just pass partial fields we need to update? UPD: just read the article, I think that's correct

Yes that's correct! 👍 

'Upsert' == update + insert. It means if the entry is found in the db, then update it; and if it's not, then insert one using that data/request body.

With a regular "update", if the entry isn't found in the db, you would just return a "not found" error and not try to now create that entry.

Reply:
> put overwrights everything so if the fields are not required and we pass less properties than it had initially- it will fully replace the item in a db and we'll have only properties we passed in our put request.

Exactly! The id of the object would still be the same; but all the fields of that object will be replaced with what was given in the request body.

Reply:
@akosasante thank you so much! Love your explanations!!!